src/build_network.py

```diff
@@ -83,15 +87,23 @@ def parse_metadata(file_path: str) -> pd.DataFrame:
     print('Building graphs...')
     # For reasons that are unclear to me, this is faster than either inserting all
     # edges at once or constructing the graph in one shot
+    # TODO save results every X headings (probably ~100 prevents OOM but is still fast)
     for file_path in tqdm(glob.glob(f'{args.data_dir}/*')):
         citation_list = pd.read_csv(file_path)
         for heading in headings:
             heading_dois = heading_to_dois[heading]
             for citing, cited in zip(citation_list['citing'], citation_list['cited']):
-                if citing in heading_dois and cited in heading_dois:
-                    heading_to_graph[heading].add_edge(citing, cited)
+                if args.include_first_degree:
```

Comment:
How much bigger does this typically make your networks? Does it vary by heading?

Reply:
Iirc the first degree nodes tends to increase the number of nodes by ~tens of percents, but pretty dramatically increases the number of edges